error when running `make scaffold-gateway` from clean repo

**Description**

When attempting to run `make scaffold-gateway` from a clean pull, the command generates an error and fails.

**Expected Behavior**

The command completes successfully

**Actual Behavior**

```
$ make scaffold-gateway
cd tools/scaffolding && go run scaffolder.go -m gateway -p 9366ad4
2020/08/14 17:12:09 Welcome!
*** Analyzing environment...
> GOPATH: /home/privateip/go
> User: privateip

*** Based on your environment, we've picked the following destination for your new repo:
> /home/privateip/go/src/github.com/privateip/clutch-custom-gateway

Note: please pay special attention to see if the username matches your provider's username.
Is this okay? [Y/n]: y

*** Generating...
2020/08/14 17:12:11 Using templates in /home/privateip/clutch/tools/scaffolding/templates/gateway
2020/08/14 17:12:11 Using tmpdir /tmp/clutch-scaffolding-346449552
2020/08/14 17:12:11 .gitattributes
2020/08/14 17:12:11 .gitignore
2020/08/14 17:12:11 Makefile
2020/08/14 17:12:11 api/echo/v1/echo.proto
2020/08/14 17:12:11 backend/.air.toml
2020/08/14 17:12:11 backend/clutch-config.yaml
2020/08/14 17:12:11 backend/cmd/assets/placeholder.go
2020/08/14 17:12:11 backend/go.mod
2020/08/14 17:12:11 backend/main.go
2020/08/14 17:12:11 backend/module/echo/echo.go
2020/08/14 17:12:11 backend/module/echo/echo_test.go
2020/08/14 17:12:11 backend/tools.go
2020/08/14 17:12:11 frontend/lerna.json
2020/08/14 17:12:11 frontend/package.json
2020/08/14 17:12:11 frontend/public/index.html
2020/08/14 17:12:11 frontend/src/clutch.config.js
2020/08/14 17:12:11 frontend/src/index.css
2020/08/14 17:12:11 frontend/src/index.jsx
2020/08/14 17:12:11 frontend/workflows/echo/.eslintrc.js
2020/08/14 17:12:11 frontend/workflows/echo/babel.config.js
2020/08/14 17:12:11 frontend/workflows/echo/jest.config.js
2020/08/14 17:12:11 frontend/workflows/echo/package.json
2020/08/14 17:12:11 frontend/workflows/echo/prettier.config.js
2020/08/14 17:12:11 frontend/workflows/echo/src/echo.jsx
2020/08/14 17:12:11 frontend/workflows/echo/src/index.jsx
2020/08/14 17:12:11 Adding clutch dependencies to go.mod...
2020/08/14 17:12:13 Generating API code from protos...
2020/08/14 17:12:13 cd /tmp/clutch-scaffolding-346449552 && make api
Downloading yarn v1.22.4 to build environment...
/usr/bin/env bash /home/privateip/go/pkg/mod/github.com/lyft/clutch/tools@v0.0.0-20200814192005-9366ad4746b6/compile-protos.sh /tmp/clutch-scaffolding-346449552
info: downloading core APIs 9366ad4746b6 to build environment...
info: Downloading protoc-v3.12.4 to build environment
info: Downloading protobufjs to build environment
yarn add v1.22.4
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved 13 new dependencies.
info Direct dependencies
└─ protobufjs@6.10.1
info All dependencies
├─ @protobufjs/aspromise@1.1.2
├─ @protobufjs/base64@1.1.2
├─ @protobufjs/codegen@2.0.4
├─ @protobufjs/eventemitter@1.1.0
├─ @protobufjs/fetch@1.1.0
├─ @protobufjs/float@1.0.2
├─ @protobufjs/path@1.1.2
├─ @protobufjs/pool@1.1.0
├─ @protobufjs/utf8@1.1.0
├─ @types/long@4.0.1
├─ @types/node@13.13.15
├─ long@4.0.0
└─ protobufjs@6.10.1
Done in 1.00s.
info: compiling go
/tmp/clutch-scaffolding-346449552/api/echo/v1
info: compiling javascript bundle
installing semver@^7.1.2
child_process.js:644
    throw err;
    ^

Error: Command failed: npm --silent install semver@^7.1.2
    at checkExecSyncError (child_process.js:601:13)
    at Object.execSync (child_process.js:641:13)
    at modInstall (/tmp/clutch-scaffolding-346449552/build/bin/protobufjs-6.10.1/node_modules/protobufjs/cli/util.js:129:19)
    at Object.exports.setup (/tmp/clutch-scaffolding-346449552/build/bin/protobufjs-6.10.1/node_modules/protobufjs/cli/util.js:140:9)
    at Object.<anonymous> (/tmp/clutch-scaffolding-346449552/build/bin/protobufjs-6.10.1/node_modules/protobufjs/cli/pbjs.js:7:6)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
Makefile:15: recipe for target 'api' failed
make[1]: *** [api] Error 1

2020/08/14 17:12:23 `make api` in the destination dir returned the above error
exit status 1
Makefile:138: recipe for target 'scaffold-gateway' failed
make: *** [scaffold-gateway] Error 1
```

**Version**
$ git rev-parse HEAD
9366ad4746b698a81635803db8addd791140f694

**Other Context**
<!-- Include any relevant configuration excerpts, etc. -->

--- a/tools/compile-protos.sh
+++ b/tools/compile-protos.sh
@@ -37,6 +37,8 @@ done
 shift "$((OPTIND-1))" # shift so that $@, $1, etc. refer to the non-option arguments
 
 main() {
+  check_prereqs
+
   REPO_ROOT="${SCRIPT_ROOT}"
   # Use alternate root if provided as command line argument.
   if [[ -n "${1-}" ]]; then
@@ -214,6 +216,13 @@ prepare_build_environment() {
 
   if [[ "${ACTION}" == "lint" ]]; then
     install_clang_format
+  fi
+}
+
+check_prereqs() {
+  if ! command -v "npm" &> /dev/null; then
+    echo "ERROR: npm not found, see https://github.com/lyft/clutch/wiki/Requirements#nodejs for more information."
+    exit 1
   fi
 }
 

